### ISSUE TITLE
docker-machine create should check for the existence of netstat

### ISSUE THREAD

**Issue by johnjelinek** [docker/machine#2481]
The error message is not very clear when you don't have netstat installed.

**Comment by johnjelinek** [comment]
This was first discovered in #2480

**Comment by nathanleclaire** [comment]
cc @docker/machine-maintainers

**Comment by treyhyde** [comment]
It would be additionally helpful if the official apt-packages listed net-tools as a dependency.

**Comment by spul** [comment]
Any reason why netstat is used in favor of ``ss``? netstat is flagged obsolete anyway:

From man netstat:
```
NOTES
       This program is obsolete.  Replacement for netstat is ss.  Replacement for netstat -r is ip route.  Replacement for netstat -i is ip -s link.  Replacement for netstat -g is ip maddr.
```

**Comment by dayreiner** [comment]
Just a note that this also breaks node creation with the softlayer driver -- at least with CentOS. The softlayer CentOS 7 images (and perhaps others) don't include the net-tools package when provisioned.

**Comment by nathanleclaire** [comment]
> Any reason why netstat is used in favor of ss? netstat is flagged obsolete anyway:

Didn't know, good callout.  It seems `ss` is available even in Tiny Core Linux (boot2docker).

**Comment by nathanleclaire** [comment]
@dayreiner @johnjelinek @treyhyde Is `ss` present on your systems which lack `netstat`?

**Comment by dayreiner** [comment]
@nathanleclaire Yes, ss looks to be included by default in IBM Softlayer's minimal CentOS image (as part of the iproute package). I expect this would be true for other distro images as well at SL. 

cc @underscorephil

**Comment by nathanleclaire** [comment]
Will take a look into using `ss` instead

**Comment by underscorephil** [comment]
Apologies for the slow response. It should be included in the other distros. If you run into any that dont have it, please reach out to me and I'll see about getting it added.

**Comment by nathanleclaire** [comment]
I'm not sure if using `ss` will work.  In the shell it is fine but I cannot invoke `docker-machine ssh name ss`, it returns error 127.  (We go over SSH to do this provisioning step, so any solution has to work over SSH)  I'm not entirely sure why.

**Comment by nathanleclaire** [comment]
Invoking `dm ssh default /usr/local/sbin/ss` works on boot2docker, but I'm not sure that will be in the same location on every distro.

**Comment by nathanleclaire** [comment]
What distro(s) is netstat missing on?  Maybe we could make sure that package is installed as well before going ahead with provisoining.

**Comment by dayreiner** [comment]
@nathanleclaire Looking at #2480, that also appears to have been CentOS (I'm guessing 7), so it may just be a Centos (and possible RHEL) thing. If I run `ssh $host ss` I get the expected output going to a Centos7 box though -- looks like on Centos its in `/usr/sbin`, doesn't seem to be consistent between distros.

Passing `yum -y install net-tools` on the first connection should be enough to ensure netstat is available though.

**Comment by spul** [comment]
Guys, installing a deprecated package is not the solution. I don't know how dm parses its command line arguments, ... In any normal *nix it's like ``ssh remote-host -- ss -lt``. I'll verify this for dm with CentOS today and get back to this.

**Comment by spul** [comment]
The issue with dm ssh machine ss is not ss being somewhere, the issue is $PATH being limited to /usr/local/bin and /usr/bin (at least with dm using ssh binary (default) in favor of Go native implementation.

```
dm -s environments/local/.docker/machine/ ssh node1
[vagrant@node1 ~]$ echo $PATH
/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/vagrant/.local/bin:/home/vagrant/bin
```

```
dm -s environments/local/.docker/machine/ ssh node1 -- 'echo $PATH'
/usr/local/bin:/usr/bin
```

If setting PATH is in dm control I do not see any reason why to limit the PATH to these two folders and use the common defaults for non-privileged users:

``/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin``
